distchan.go

```diff
@@ -81,17 +121,14 @@ func (s *Server) AddDecoder(f Transformer) *Server {
 
 // Ready returns true if there are any clients currently connected.
 func (s *Server) Ready() bool {
-	s.mu.RLock()
-	defer s.mu.RUnlock()
-	return len(s.connections) > 0
+	return atomic.LoadInt32(&s.conncnt) > 0
 }
 
 // WaitUntilReady blocks until the server has at least one client available.
 func (s *Server) WaitUntilReady() {
 	for {
-		time.Sleep(1 * time.Second)
+		runtime.Gosched()
```

Comment:
Excellent call here; I had originally written this method as a quick way to get the example working, and hadn't revisited it since then.